qdrant-landing/content/documentation/tutorials/retrieval-quality.md

```diff
@@ -0,0 +1,221 @@
+metrics are, in turn, based on the relevance score of the documents.
+
+If we treat the search pipeline as a whole, we could use them all. The same is true for the embeddings quality evaluation. However, for the 
+ANN algorithm itself, anything based on the relevance score or ranking is not applicable. Ranking in vector search relies on the distance
+between the query and the document in the vector space, however distance is not going to change due to approximation, as the function is
+still the same. 
+
+Therefore, it only makes sense to measure the quality of the ANN algorithm by the number of relevant documents in the top-k search results, 
+such as `precision@k`. It is calculated as the number of relevant documents in the top-k search results divided by `k`. In case of testing
+just the ANN algorithm, we can use the exact kNN search as a ground truth, with `k` being fixed. It will be a measure on **how well the ANN
+algorithm approximates the exact search**.
+
+## Measure the quality of the search results
+
+Let's build a quality evaluation of the ANN algorithm in Qdrant. We will, first, call the search endpoint in a standard way to obtain
+the approximate search results. Then, we will call the exact search endpoint to obtain the exact matches, and finally compare both results
+in terms of precision.
+
+Before we start, let's create a collection, fill it with some data and then start our evaluation. We will use the same dataset as in the
+[Loading a dataset from Hugging Face hub](/documentation/tutorials/huggingface-datasets/) tutorial, `Qdrant/arxiv-titles-instructorxl-embeddings`
+from the [Hugging Face hub](https://huggingface.co/datasets/Qdrant/arxiv-titles-instructorxl-embeddings). Let's download it in a streaming
+mode, as we are only going to use part of it.
+
+```python
+from datasets import load_dataset
+
+dataset = load_dataset(
+    "Qdrant/arxiv-titles-instructorxl-embeddings", split="train", streaming=True
+)
+```
+
+We need some data to be indexed and another set for the testing purposes. Let's get the first 50000 items for the training and the next 1000
+for the testing.
+
+```python
+dataset_iterator = iter(dataset)
+train_dataset = [next(dataset_iterator) for _ in range(60000)]
+test_dataset = [next(dataset_iterator) for _ in range(1000)]
+```
+
+Now, let's create a collection and index the training data. This collection will be created with the default configuration. Please be aware that
+it might be different from your collection settings, and it's always important to test exactly the same configuration you are going to use later
+in production.
+
+```python
+from qdrant_client import QdrantClient, models
+
+client = QdrantClient("http://localhost:6333")
+client.create_collection(
+    collection_name="arxiv-titles-instructorxl-embeddings",
+    vectors_config=models.VectorParams(
+        size=768,  # Size of the embeddings generated by InstructorXL model
+        distance=models.Distance.COSINE,
```

Comment:
It would be nice for someone to understand how different distance algorithms affect search results, I am sure, that falls under the measure of retrieval quality too.

Reply:
Good idea, thanks! How about making a note about this?

![](https://github.com/qdrant/landing_page/assets/2649301/b35c6829-5398-4f46-b866-8ddfcfe52122)

Reply:
Looks good ! Thanks !